packages/libs/src/services/hedgehog/Hedgehog.ts

```diff
@@ -14,6 +14,41 @@ export type HedgehogConfig = {
   localStorage?: LocalStorage
   createKey?: CreateKey
 }
+
+const walletEventMethods = [
+  'signUp',
+  'login',
+  'logout',
+  'resetPassword',
+  'changeCredentials',
+  'createWalletObj'
+] as const
+
+// Utility function to wrap methods on HedgehogBase that modify the wallet
+// and dispatch an event afterwards
+function wrapAndNotify(
```

Comment:
I am continuing to add dark magic to this class. But it's short-lived until we can rip it out of libs :-)